src/lib/auth.ts

```diff
@@ -165,6 +164,14 @@ export const authenticate = (devMode: boolean) => {
 
             resolve(user);
           });
+      })
+      .catch((e) => {
+        reject(e);
       });
   });
 };
+
+export const _private = {
```

Comment:
Here's a go at private variables exported for testing

Reply:
In a previous PR of mine, Eli raised the question of whether parts of the interface being exported specifically for testing should be identified as not truly being part of the public interface. There was a slack discussion about it and one of the proposed solutions Eli found involved a convention of encapsulating them in a `__private` member. We can decide how we feel about it.

Reply:
For this PR I think I liked it - it's very easy to add the `_private` member and you don't end up polluting auto-complete options. I also considered naming private methods something like `_foo` but that makes them harder to use internally.

Reply:
If a private variable/method is truly private it should be marked private and thus is not available for testing which is fine if there is a test path that can reach the code via the public api.

If you are worried that consumers of the module could change the value of a variable mark it constant if it is a constant or use a private variable in the module and expose a public getter for testing for private data that is variable but needs to be exposed to tests.

Reply:
re: paragraph 1 - @kswenson brought up a point that I agree with: it's nicer to be able to test against constants pulled from files than constants written by hand. So if I want to make sure that the dev user is named `Sofia` I'd rather pull that out of `auth.ts` than check against the string `Sofia`, which makes my tests brittle. So I need some way of accessing private members in tests. 

Really, my main concern is polluting APIs/auto-completions. Someone could try changing `DEV_USER.lastName`, which would be solved by public getters, but I'm less worried about that.